spring-context/src/main/java/org/springframework/scheduling/annotation/AbstractAsyncConfiguration.java

```diff
@@ -81,7 +81,8 @@ void setConfigurers(ObjectProvider<AsyncConfigurer> configurers) {
 			}
 			return candidates.get(0);
 		});
-		this.executor = adapt(configurer, AsyncConfigurer::getAsyncExecutor);
+		Supplier<Executor> adaptedExecutor = adapt(configurer, AsyncConfigurer::getAsyncExecutor);
+		customizers.ifAvailable(customizer -> this.executor = () -> customizer.customize(adaptedExecutor.get()));
```

Comment:
`1. this.executor` won't be set if customizers aren't available. Is this desired ?
2. customizers has plural form but in fact it's a single customizer, not a collection

Reply:
Nope, good catch